hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieCreateRecordUtils.scala

```diff
@@ -143,9 +143,12 @@ object HoodieCreateRecordUtils {
               avroRecWithoutMeta
             }
 
-            val hoodieRecord = if (shouldCombine && !precombineEmpty) {
-              val orderingVal = HoodieAvroUtils.getNestedFieldVal(avroRec, precombine,
-                false, consistentLogicalTimestampEnabled).asInstanceOf[Comparable[_]]
+            //TODO [HUDI-8574] we can throw exception if field doesn't exist
+            // lazy so that we don't evaluate if we short circuit the boolean expression in the if below
+            lazy val orderingVal = HoodieAvroUtils.getNestedFieldVal(avroRec, precombine,
+              true, consistentLogicalTimestampEnabled).asInstanceOf[Comparable[_]]
```

Comment:
do we really need lazy here. whats the issue w/ previous code here? may be I am missing something.